src/common/components/auth/SessionStatusModal.jsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import { BodyLong, Button, Heading, Modal } from "@navikt/ds-react";
+import LoginBubble from "@/src/common/components/images/LoginBubble";
+
+function SessionStatusModal({ markAsLoggedOut, setHasBeenLoggedIn, login, logout, timeoutLogout, hasBeenLoggedIn }) {
+    const [isSessionExpiring, setIsSessionExpiring] = useState(null);
+    const [isSessionTimingOut, setIsSessionTimingOut] = useState(null);
+    const [isTimeoutModalOpen, setIsTimeoutModalOpen] = useState(false);
+
+    const handleSessionInfoResponse = async (response, isCurrentlyLoggedIn, errorMessage) => {
+        if (response.status === 401) {
+            markAsLoggedOut();
+            if (isCurrentlyLoggedIn) {
+                setHasBeenLoggedIn(false);
+                timeoutLogout();
+            }
+        } else if (response.status < 200 || response.status >= 300) {
+            console.error(errorMessage);
```

Comment:
Burde vi også formidle denne feilen til brukeren, i stedet for console?

Reply:
Vi har egentlig kommet frem til nei i første omgang, om det skjer noe galt her så vil brukeren enten logges ut dersom sesjonen ikke kan verifiseres (og dermed "av sikkerhetsårsaker er du logget ut"), eller så prøves det på nytt. Men vi tenker det er fint å ha den i loggen i tilfelle vi må gjøre noe feilsøking.